include/lbann/execution_contexts/execution_context.hpp

```diff
@@ -146,9 +146,9 @@ class execution_context {
 
 private:
   /** Pointer to the training context (execution environment) for the training algorithm */
-  trainer& m_trainer;
+  trainer* m_trainer;
 
-  training_algorithm& m_training_algorithm;
+  training_algorithm* m_training_algorithm;
```

Comment:
I presume Cereal struggles with serializing reference members? In the future we may be want to replace the raw pointers with `std::shared_ptr`, e.g. if we use Cereal to checkpoint the execution context. That might be overkill though, since the execution context belongs to the trainer, which can set these fields in its setup phase.

Reply:
No, reference members should not be used in general. Pointers are superior in every way.